python/pyspark/sql/tests/connect/client/test_artifact.py

```diff
@@ -294,6 +295,15 @@ def func(x):
             self.spark.addArtifacts(file_path, file=True)
             self.assertEqual(self.spark.range(1).select(func("id")).first()[0], "Hello world!!")
 
+    def test_cache_artifact(self):
```

Comment:
@HyukjinKwon For some unknown reasons the test suit freezes after the test completes successfully.

I have added similar test for Scala client, please, review it: https://github.com/apache/spark/pull/41493